udpa/extension/v1/extension.proto

```diff
@@ -0,0 +1,24 @@
+syntax = "proto3";
+
+package udpa.extension.v1;
```

Comment:
Should `TypedExtensionConfig` be in the `xds.extension` tree?  Or should it be in `xds.core`?  This isn't itself an extension; it's really a type used to point at an extension, which is a capability that is basically built into core.